docs/rfc/rfc-020-onboarding-projects.rst

```diff
@@ -0,0 +1,235 @@
+=======================================
+RFC 020: Tendermint Onboarding Projects
+=======================================
+
+.. contents::
+   :backlinks: none
+
+Changelog
+---------
+
+- 2022-03-30: Initial draft. (@tychoish)
+- 2022-04-25: Imported document to tendermint repository. (@tychoish)
+
+Overview
+--------
+
+This document describes a collection of projects that might be good for new
+engineers joining the Tendermint Core team. These projects mostly describe
+features that we'd be very excited to see land in the code base, but that are
+intentionally outside of the critical path of a release on the roadmap, and
+have the following properties that we think make good on-boarding projects:
+
+- require relatively little context for the project or its history beyond a
+  more isolated area of the code.
+
+- provide exposure to different areas of the codebase, so new team members
+  will have reason to explore the code base and gain expeirence with more than
+  one area of the system.
+
+- be of moderate size, thereby avoiding making small trivial or mechanical changes
+  while also avoiding large intractable changes without natural touchpoints or
+  check-ins.
```

Comment:
A suggestion based on what I think is the intended spirit here (with which I agree):
```suggestion
- be of moderate size, striking a healthy balance between trivial or mechanical
  changes (which provide little insight) and large intractable changes that require
  deeper insight than is available during onboarding to address well. A good size
  project should have natural touchpoints or check-ins.
```

"Check-ins" makes a nice litmus test for both ends: Boring changes don't require much feedback, because they're boring and obvious and can be validated by just doing the normal engineering things (writing tests, making sure stuff doesn't catch fire). Systemic changes lack easy feedback because there's too much unknown.

So, yeah, I really like that rubric.